src/core/utils/multicall.ts

```diff
@@ -89,28 +94,67 @@ export class MulticallQueue {
       const batchSize = callsToProcess.length
       logger.debug({ batchSize }, 'Sending multicall batch ')
 
-      const results = await this.multicallContract.aggregate3.staticCall(
-        callsToProcess.map(({ callParams }) => callParams),
-        {
-          blockTag: blockTag == LATEST ? undefined : blockTag,
-        },
-      )
+      let results: Multicall3.ResultStructOutput[]
+      try {
+        results = await this.multicallContract.aggregate3.staticCall(
+          callsToProcess.map(({ callParams }) => callParams),
+          {
+            blockTag: blockTag == LATEST ? undefined : blockTag,
+          },
+        )
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      } catch (error: any) {
+        callsToProcess.forEach(({ reject }) => {
+          reject(
+            new MulticallError({
+              message: 'RPC provider error',
+              chainId: this.chainId,
+              flushTimeoutMs: this.flushTimeoutMs,
+              maxBatchSize: this.maxBatchSize,
+            }),
+          )
+        })
+
+        logger.error(
+          {
+            chainId: this.chainId,
+            flushTimeoutMs: this.flushTimeoutMs,
+            maxBatchSize: this.maxBatchSize,
+            batchSize,
+            error: error?.info?.error,
+          },
+          'Multicall error when sending a batch',
+        )
+
+        return
+      }
 
       const resultLength = results.length
 
       if (resultLength !== batchSize) {
```

Comment:
We shouldn't remove this, as I actually managed to trigger it multiple times by increasing the batch size to several thousands.

Reply:
Oh wow. Thats odd, i didnt think it would be possible